File: C#/Sliding Window/1004. Max Consecutive Ones III.cs
/* 1004. Max Consecutive Ones III 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/max-consecutive-ones-iii/submissions/1391227724/ */

public class Solution {
    public int LongestOnes(int[] nums, int k) {
        int zero_cnt = 0, l = 0, ans = 0;
        for (int r = 0; r < nums.Length; ++r) {
            if (nums[r] == 0)
                zero_cnt++;
            while (zero_cnt > k)
                if (nums[l++] == 0)
                    zero_cnt--;
            ans = Math.Max(ans, r - l + 1);
        }
        return ans;
    }
}

File: JavaScript/Sliding Window/1004. Max Consecutive Ones III.js
/* 1004. Max Consecutive Ones III 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/max-consecutive-ones-iii/submissions/1391225265/ */

var longestOnes = function (nums, k) {
    let l = max_num = 0

    for (let r = 0; r < nums.length; r++) {
        k -= 1 - nums[r]

        if (k < 0) {
            k += 1 - nums[l]
            l++
        } else {
            max_num = Math.max(max_num, r - l + 1)
        }
        console.log(l, r)
    }

    return max_num
};